pkg/kubelet/apis/config/types.go

```diff
@@ -407,6 +407,16 @@ type KubeletConfiguration struct {
 	// class value that lies in the range of that value and the next higher entry in the
 	// list when the node is shutting down.
 	ShutdownGracePeriodByPodPriority []ShutdownGracePeriodByPodPriority
+	// ShutdownSetPodsPhaseToTerminal sets pods to terminal phase during shutdown (enabled by default).
+	// If a node becomes ready again after graceful node shutdown (e.g. a reboot),
+	// then having ShutdownSetPodsPhaseToTerminal enabled, will result in pods not being started on the same node
+	// since the pods will be placed into terminal phase.
+	// In contrast, if ShutdownSetPodsPhaseToTerminal is disabled, pods will not be placed into terminal phase,
+	// and as result they will resume to run after the node becomes ready again.
+	// Default: true
+	// +featureGate=GracefulNodeShutdown
+	// +optional
+	ShutdownSetPodsPhaseToTerminal bool
```

Comment:
Thanks @liggitt, makes sense to make this more extensible in case we want to add more options in the future.

> Does the current behavior just set terminal, or does it also set a deletionTimestamp on the pod? If the former, is setting a deletionTimestamp another possibility that should be represented?

The former only sets phase to terminal, ti does not set deletion timestamp. But I agree, it makes sense make this a string with policy types incase we want to add something down the road.

Reply:
The thought had also occurred to me but I wasn't sure if it'd be blocking. Agreed!